SU2_CFD/src/solvers/CSolver.cpp

```diff
@@ -3327,9 +3335,87 @@ void CSolver::LoadInletProfile(CGeometry **geometry,
   if (time_stepping)
     profile_filename = config->GetUnsteady_FileName(profile_filename, val_iter, ".dat");
 
+ 
+  // create vector of column names
+  for (iMarker = 0; iMarker < config->GetnMarker_All(); iMarker++) {
+
+    /*--- Skip if this is the wrong type of marker. ---*/
+    if (config->GetMarker_All_KindBC(iMarker) != KIND_MARKER) continue;
+
+    string Marker_Tag = config->GetMarker_All_TagBound(iMarker);
+    su2double p_total   = config->GetInlet_Ptotal(Marker_Tag);
+    su2double t_total   = config->GetInlet_Ttotal(Marker_Tag);
+    auto flow_dir = config->GetInlet_FlowDir(Marker_Tag);
+    std::stringstream columnName,columnValue;    
+
+    columnValue << setprecision(15);
+    columnValue << std::scientific;
+
+    columnValue << t_total << "\t" << p_total <<"\t";
+    for (unsigned short iDim = 0; iDim < nDim; iDim++) {
+      columnValue << flow_dir[iDim] <<"\t";
+    }
+
+    columnName << "# COORD-X  " << setw(24) << "COORD-Y    " << setw(24);
+    if(nDim==3) columnName << "COORD-Z    " << setw(24); 
+    
+    if (compressible){
```

Comment:
This is ok for now, but if more solvers start using inlet profiles this is not a scalable solution.
The suggestion was to have this block of code as a method of different solvers, so that you do not have to figure out the type of solver based on configuration options (and consequently the type of inlet, etc. etc.).
```solver->GetInletVariableNames()``` something like that.
The motivation for polymorphism and abstract classes is to allow centralizing this type of knowledge, i.e. "what type of object is this?", on a single place, the instantiation of the object.
Enough philosophy...

Reply:
yes, I prefer having something like this at the level of the solver container so we can construct this information. After all, very soon we will have passive scalar, combustion variables, nemo combustion variables,...